absl/base/attributes.h

```diff
@@ -125,13 +125,19 @@
     (defined(__GNUC__) && !defined(__clang__))
 #define ABSL_ATTRIBUTE_ALWAYS_INLINE __attribute__((always_inline))
 #define ABSL_HAVE_ATTRIBUTE_ALWAYS_INLINE 1
+#elif defined(_MSC_VER)
+#define ABSL_ATTRIBUTE_ALWAYS_INLINE __forceinline
```

Comment:
I'm really sorry about this, but I don't know of any way we are going to be able to accept the changes in `attributes.h` in the short term.

The problem is we have a lot of code that uses `ABSL_ATTRIBUTE_ALWAYS_INLINE`, but MSVC is a lot more picky about the placement of `ABSL_ATTRIBUTE_ALWAYS_INLINE` than other compilers. For example, we have code that places this attribute at the end of the declaration, and this would break that code when it is compiled with MSVC. I'm not talking about just the Abseil project, but literally the billions of lines of Google code that depend on this.

I tried to see how easy it would be to fix all the places in the code that were using this in an incompatible way, but there were too many of them. We should be able to write some kind of clang-based tool to fix them all automatically, but it isn't too high on our priority list.

Anyway, I would like to accept this change some day, but today is not that day :-(

Reply:
I understand. I will remove this from the PR.